pyproject.toml

```diff
@@ -44,7 +44,7 @@ json = [
   "pandas",
   "pydantic",
   "pint",
-  "torch",
+  "torch",  # Note: need torch>=2.3.0 for numpy 2 # 719
```

Comment:
I decide to drop this pin entirely because I believe it's something `torch` should resolve not `monty` (i.e. `torch` should correctly list supported `numpy` for dependency resolver) 

Unless `tensor.numpy()` should not be used at certain versions? I'm not a `torch` user, input appreciated!